### DISCUSSION TITLE
TDD learning resources - as recommended by YOU

### DISCUSSION THREAD

**Discussion by ngrahovac** [techexcellenceio/techexcellence#39] [General]
Hello, everyone! Glad to be a part of this community.

I'm curious to hear about your recommendations for learning about TDD - books, learning paths, etc. for a beginner. 

When it comes to testing in general, my current approach is aligned to that of Vladimir Khorikov, as outlined in his [book](https://www.amazon.com/gp/product/1617296279/ref=as_li_tl?ie=UTF8&camp=1789&creative=9325&creativeASIN=1617296279&linkCode=as2&tag=vkhorikov-20&linkId=2081de4b1cb7564cb9f95526533c3dae). I would like to hear your suggestions about where to go from here. Do not hesitate to elaborate on your choices in detail!

My next choice of literature was supposed to be Kent Beck's book, but I'd be more than happy to change my plans after hearing from you.

If the discussion gains enough traction, we could come up with a learning path, as recommended by the community.

Thanks!

**Comment by valentinajemuovic** [comment]
Kent Beck's book & Uncle Bob Cleaner Coders - Classicist TDD
Steve Freeman's GOOS book - Mockist TDD

Also C++ TDD book which I discovered after @mica16 highly recommended it on LinkedIn + meetups as the best book https://www.amazon.com/Modern-Programming-Test-Driven-Development-Better/dp/1937785483 and I must agree it's much better organized than the older ones

**Comment by biomed12** [comment]
I have read the two books at least twice and they are wonderful for me:

-https://www.amazon.com/Driven-Development-Embedded-Pragmatic-Programmers/dp/193435662X
-https://www.amazon.com/Modern-Programming-Test-Driven-Development-Better/dp/1937785483 (Valentina has already recommended)

**Comment by marciovmartins** [comment]
Besides books, try some coding dojo groups. https://codingdojo.org/

Very useful hands-on practice and helped me a lot to be proficient doing TDD
